awot/io/read_hiaper_radar.py

```diff
@@ -0,0 +1,214 @@
+        platform : str
+        Platform name or identifier.
+        intrument_name: str
+        Name of the instrument HCR
+        '''
+    # Create a dictionary to hold data
+    data = {}
+    
+    # Read the NetCDF
+    ncFile = Dataset(fname, 'r')
+    ncvars = ncFile.variables
+    
+    # Grab the metadata stored in global attributes as a dictionary
+    try:
+        data['metadata'] = ncFile.__dict__
+    except:
+        data['metadata'] = None
+    
+    try:
+        data['project'] = ncFile.institution
+    except:
+        data['project'] = None
+
+    try:
+        data['flight_number'] = ncFile.start_datetime
+    except:
+        data['flight_number'] = None
+    
+    # Set the platform
+    try:
+        data['platform'] = ncFile.site_name
+    except:
+        data['platform'] = 'GV'
+
+    try:
+        data['intrument_name'] = ncFile.intrument_name
+    except:
+        data['platform'] = 'HCR'
+
+
+    # Find the indices of not missing points
+    Good = np.where(~np.isnan(ncFile.variables['time'][:]))
+
+    data['time'] = _get_time(fname, ncFile, Good)
+    
+    # Grab the name map
+    name_map_data = _get_hcr_data_map()
+    
+    # Loop through the variables and pull data
+    for varname in name_map_data:
+        if name_map_data[varname] in ncvars:
+            data[varname] = common._ncvar_to_dict(
+                                                  ncvars[name_map_data[varname]])
+        else:
+            data[varname] = None
+            common._var_not_found(varname)
+    # Add fields to their own dictionary
+    fields = {}
+
+    # Grab a name map for HCR field data
+    name_map_fields = _get_hcr_field_name_map()
+    
+    # Loop through the variables and pull data
+    for varname in name_map_fields:
+        if name_map_fields[varname] in ncvars:
+            fields[varname] = common._ncvar_subset_to_dict(
+                                                           ncvars[name_map_fields[varname]], Good)
+        else:
+            fields[varname] = None
+            common._var_not_found(varname)
+
+    #cant remember do we need this SURFACE DATA?
+
+    # Find the surface variable
+    # See PROCESSED_DATA/WCR_L2_OWLES13.20131015.cdl
+    # for details of mask properties
+    surface = np.empty_like(data['latitude']['data'])
+    condition = np.equal(fields['mask']['data'], 32)
+    for nn in range(len(surface)):
+        if np.any(condition[nn, :]):
+            surface[nn] = data['height']['data'][np.where(
+                                                          condition[nn, :])[0][0]]
+    data['surface'] = {'name': "surface",
+        'long_name': "Height of Surface",
+            'data': surface,
+                'units': 'meters'
+                }
+
+    # Save to output dictionary
+    data['fields'] = fields
+        
+        # Set the data format
+    try:
+        data['data_format'] = ncFile.version
+    except:
+        data['data_format'] = 'CF-Radial-1.3'
+        
```

Comment:
Let's use something like 'hcr_vertical' for the **data_format** key. This label is just an internal way to keep track of the type of file we are dealing with and not the true data format. You can check out some of the other readers for examples.